hamt/shardeddir.go

```diff
@@ -256,7 +256,14 @@ func (itr *_UnixFSShardedDir__ListItr) Done() bool {
 // can be expected that itr.Next will be called node.Length times
 // before itr.Done becomes true.
 func (n UnixFSHAMTShard) ListIterator() ipld.ListIterator {
```

Comment:
I wonder if we want to do this as a list -

The IPLD node interface for a directory will be of kind `Map`, so by the IPLD contract i would expect that asking for a list iterator returns `nil.

A non-ipld interface + method might be clearer for accessing this functionality.